src/test/java/seedu/address/logic/commands/DeleteTagCommandTest.java

```diff
@@ -0,0 +1,140 @@
+package seedu.address.logic.commands;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+import static seedu.address.logic.commands.CommandTestUtil.assertCommandFailure;
+import static seedu.address.logic.commands.CommandTestUtil.assertCommandSuccess;
+import static seedu.address.testutil.TypicalParcels.getTypicalAddressBook;
+
+import java.util.Iterator;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import javafx.collections.ObservableList;
+import seedu.address.commons.exceptions.IllegalValueException;
+import seedu.address.logic.CommandHistory;
+import seedu.address.logic.UndoRedoStack;
+import seedu.address.model.Model;
+import seedu.address.model.ModelManager;
+import seedu.address.model.UserPrefs;
+import seedu.address.model.parcel.ReadOnlyParcel;
+import seedu.address.model.tag.Tag;
+import seedu.address.model.tag.exceptions.TagNotFoundException;
+
+
+/**
+ * Contains integration tests (interaction with the Model) and unit tests for ListCommand.
```

Comment:
Please separate out your integration command test

Reply:
For this, I think it's better to keep it inside.